compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -1120,7 +1120,7 @@ pub trait PrettyPrinter<'tcx>:
         //
         // To avoid that causing instabilities in compiletest
         // output, sort the auto-traits alphabetically.
-        auto_traits.sort();
+        auto_traits.sort_by_key(|(name, _did)| name.clone());
```

Comment:
```suggestion
        auto_traits.sort_by_cached_key(|(name, _did)| name.clone());
```

Reply:
In other hand, we can:
```rust
let mut auto_traits: Vec<_> = predicates.auto_traits().collect();
auto_traits.sort_by_cached_key(|did| self.tcx().def_path_str(did));
```

Reply:
The `sort_by_cached_key(|did| self.tcx().def_path_str(did))` version seems to looks the most clean.